packages/math/src/pool/concentrated/quotes.ts

```diff
@@ -0,0 +1,112 @@
+import { Coin, Dec, Int } from "@keplr-wallet/unit";
+
+import { maxSpotPrice, minSpotPrice, smallestDec } from "./const";
+import { TickOverflowError } from "./errors";
+import { makeSwapStrategy } from "./swap-strategy";
+import { sqrtPriceToTick, tickToSqrtPrice } from "./tick";
+import { addLiquidity, approxSqrt } from "./utils";
+
+export const ConcentratedLiquidityMath = {
+  calcOutGivenIn,
+  // calcInGivenOut,
+  // calcSpotPrice,
+};
+
+interface SwapState {
+  amountRemaining: Dec;
+  amountCalculated: Dec;
+  sqrtPrice: Dec;
+  tick: Int;
+  liquidity: Dec;
+  feeGrowthGlobal: Dec;
+}
+
+function calcOutGivenIn(
+  tokenIn: Coin,
+  tokenDenom0: string,
+  poolLiquidity: Dec,
+  tickDepths: Int[],
+  curTickIndex: number,
+  curTick: Int,
+  curSqrtPrice: Dec,
+  precisionFactorAtPriceOne: number,
+  swapFee: Dec,
+  priceLimit = new Dec(0)
+): Int {
+  const tokenInAmountSpecified = new Dec(tokenIn.amount);
+
+  const isZeroForOne = tokenIn.denom === tokenDenom0;
+  if (isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = minSpotPrice;
+  } else if (!isZeroForOne && priceLimit.equals(new Dec(0))) {
+    priceLimit = maxSpotPrice;
+  }
+
+  const sqrtPriceLimit = approxSqrt(priceLimit); // TODO: use function on updated keplr unit package: https://github.com/chainapsis/keplr-wallet/pull/674
+
+  const swapStrategy = makeSwapStrategy(
+    isZeroForOne,
+    true,
+    sqrtPriceLimit,
+    swapFee
+  );
+
+  const swapState: SwapState = {
+    amountRemaining: tokenInAmountSpecified,
+    amountCalculated: new Dec(0),
+    sqrtPrice: curSqrtPrice,
+    tick: swapStrategy.initTickValue(curTick) ?? curTick,
+    liquidity: poolLiquidity,
+    feeGrowthGlobal: new Dec(0),
+  };
+
+  let sqrtPriceStart: Dec;
+  let i = curTickIndex;
+  while (
+    swapState.amountRemaining.gt(smallestDec) &&
+    !swapState.sqrtPrice.equals(sqrtPriceLimit)
+  ) {
+    sqrtPriceStart = swapState.sqrtPrice;
+
+    const nextTick: Int | undefined = tickDepths?.[i];
```

Comment:
@p0mvn thoughts on these `Int` depth values for ticks? Obv since we don't deal with stores directly on frontend, we just use an array of depths (originating from this query https://app.clickup.com/t/860pq0mn3). Does this check out? Then, later I throw an exception (`TickOverflowError`) if we ran out of ticks, indicating we need to query again for more ticks.

Reply:
Discussing this on Slack but, basically, considering changing this from "tick depth" semantics to "liquidity net" model that is aligend with swap computations chain side.

The query is still tracked by that issue and the suggestion to utilize "liquidity net' does not affect the progress we've already made on it. 

The use of arrays makes sense to me.

Let's just get an agreement on "tick depth" vs "liquidity net"